daemons/ipa-kdb/ipa_kdb_mspac.c

```diff
@@ -1835,27 +1840,60 @@ krb5_error_code filter_logon_info(krb5_context context,
     /* check netbios/flat name */
     if (strcasecmp(info->info->info3.base.logon_domain.string,
                    domain->flat_name) != 0) {
-        krb5_klog_syslog(LOG_ERR, "PAC Info mismatch: domain = %s, "
-                                  "expected flat name = %s, "
-                                  "found logon name = %s",
-                                  domain->domain_name, domain->flat_name,
-                                  info->info->info3.base.logon_domain.string);
-        return EINVAL;
+        struct ipadb_mspac *mspac_ctx = ipactx->mspac;
+        result = FALSE;
+        /* Didn't match but perhaps the original PAC was issued by a child domain's DC? */
+        if ((mspac_ctx != NULL) && (domain->parent == NULL)) {
+            /* this is forest root, search for any domains under it */
+            for (k = 0; k < mspac_ctx->num_trusts; k++) {
+                if ((mspac_ctx->trusts[k].parent == domain) &&
+                    strcasecmp(info->info->info3.base.logon_domain.string,
+                               mspac_ctx->trusts[k].flat_name) == 0) {
+                    result = TRUE;
+                    break;
```

Comment:
Would it be helpful for troubleshooting to log the domains that don't match or is that already obvious from the PAC logged elsewhere?

Reply:
We already record that when nothing matches, this is what triggered the rework.

However, I think we have to drop this particular check. In MIT Kerberos 1.20 an interface that calls into `issue_pac()` driver's callback did change. We don't receive an original client Kerberos principal anymore. Instead, we get a DB entry associated with the client principal, if that exists. If it is not, we get signing tgt key. And here is the problem: for trusted domains' principals we only get the signing tgt key which is `krbtgt/OUR.DOMAIN@TRUSTED.DOMAIN` so all we know is the realm of the trusted forest root domain. If the original client principal was from the same forest's child domain, we may never know it as we have no Kerberos client principal.

This check needs to only apply to our own domain. For trusted domains we need to perform PAC analysis but we cannot cross-reference it anymore with the client principal.

I'll rewrite the code.